### PR TITLE
Futures market management tests

### PR DESCRIPTION
**Pull request by zyzek**
This adds a number of tests and implements a couple of fixes in the `FuturesMarketManager` contract.

### CODE REVIEW

File: test/contracts/FuturesMarketManager.js [hunk 1]
```diff
@@ -1,10 +1,367 @@
...
+			);
+		});
+
+		it('Aggregated debt updates properly as the debt values change', async () => {
+			assert.bnEqual((await futuresMarketManager.totalDebt())[0], individualDebt.mul(toBN(3)));
+			await ms[0].setMarketDebt(toUnit('2500'));
+			await ms[1].setMarketDebt(toUnit('200'));
+			assert.bnEqual(
+				(await futuresMarketManager.totalDebt())[0],
+				individualDebt.div(toBN(10)).mul(toBN(37))
```
Comment (liamzebedee) [inline test/contracts/FuturesMarketManager.js:332]: It's a little unclear why you're dividing by 10, multiplying by 37 here? Why not just write toBN(3700)?

**Review by liamzebedee — APPROVED** [review]
Just a bunch of tiny code nitpicks. Nice straightforward code, Anton.

File: test/contracts/FuturesMarketManager.js [hunk 2]
```diff
@@ -1,10 +1,367 @@
...
+				abi: futuresMarketManager.abi,
+				ignoreParents: ['Owned', 'MixinResolver', 'Proxyable'],
+				expected: ['addMarkets', 'removeMarkets', 'removeMarketsByAsset', 'issueSUSD', 'burnSUSD'],
+			});
+		});
+	});
+
+	describe('Market management', () => {
+		const currencyKeys = ['sBTC', 'sETH'].map(toBytes32);
+		let ms, addresses;
```
[the remark this answers is not in the document]
Comment (zyzek) [reply test/contracts/FuturesMarketManager.js:64]: Good call. I will resolve this.

File: test/contracts/FuturesMarketManager.js [hunk 3]
```diff
@@ -1,10 +1,367 @@
...
+			await assert.revert(market.burnSUSD(owner, toUnit('1')), 'SafeMath: subtraction overflow');
+		});
+
+		it('only markets are permitted to issue or burn sUSD', async () => {
+			await onlyGivenAddressCanInvoke({
+				fnc: futuresMarketManager.issueSUSD,
+				args: [owner, toUnit('1')],
+				accounts,
+				skipPassCheck: true,
+				reason: 'Sender is not a market',
```
[the remark this answers is not in the document]
Comment (zyzek) [reply test/contracts/FuturesMarketManager.js:293]: Given the inherent ambiguity in the proxy architecture of who exactly the sender is, I've removed the sender language altogether in favour of `Permitted only for markets`.

File: test/contracts/FuturesMarketManager.js (hunk 1 above)
Comment (zyzek) [reply test/contracts/FuturesMarketManager.js:332]: You're right, that's definitely clearer.

File: test/contracts/FuturesMarketManager.js [hunk 4]
```diff
@@ -1,10 +1,367 @@
...
+			await futuresMarketManager.addMarkets([m.address], { from: owner });
+			assert.bnEqual(await futuresMarketManager.numMarkets(), toBN(3));
+			assert.equal((await futuresMarketManager.markets(2, 1))[0], m.address);
+
+			assert.equal(await futuresMarketManager.marketForAsset(toBytes32('sLINK')), m.address);
+		});
+
+		it('Adding multiple markets', async () => {
+			const keys = ['sLINK', 'sSNX'].map(toBytes32);
+			const markets = await Promise.all(
```
[the remark this answers is not in the document]
Comment (zyzek) [reply test/contracts/FuturesMarketManager.js:102]: Yeah the snapshot occurs before and after each`it` block. If it did not, then this test would fail.

**Review by liamzebedee — APPROVED** [review]
Lookin' good!

**Pull request merged**